fastlane/fastlane
# [spaceauth] add `--copy_to_clipboard` option to `spaceauth`

### Checklist

### Motivation and Context
Resolves https://github.com/fastlane/fastlane/discussions/17843

### Description

After investigating this issue, I tried to accomplish what both users in https://github.com/fastlane/fastlane/discussions/17843 were seeking (`eval $(fastlane spaceauth --eval)`). However, that wasn't possible exactly as is because it would require _fastlane_ to not print anything else to stdout except the export command, and _fastlane_ doesn't have a global `--silent` argument or anything similar - it may print out session-related logs not directly related to spaceauth, or even the initial `[✔] 🚀`.
To work around this issue, the solution I came up with was to export the session to the clipboard without prompting the user for it, specially in a CI environment.

### Documentation

I've documented this new option in `spaceship/README.md` (in this PR) and also here https://github.com/fastlane/docs/pull/1049

### Testing Steps

To test this branch, modify your Gemfile as:

```ruby
gem "fastlane", :git => "https://github.com/fastlane/fastlane.git", :branch => "origin/rogerluan/spaceauth-export-to-clipboard"
```

And run `bundle install` to apply the changes.

Then run `bundle exec fastlane spaceauth -u user@example.org --exports_to_clipboard && eval $(pbpaste)` and `echo $FASTLANE_SESSION` to see if the session was correctly exported.

cc @return-main @Semmu
